include/aws/io/retry_strategy.h

```diff
@@ -0,0 +1,175 @@
+#ifndef AWS_IO_CLIENT_RETRY_STRATEGY_H
+#define AWS_IO_CLIENT_RETRY_STRATEGY_H
```

Comment:
This seems like a better fit for `aws-c-ops` than `aws-c-io`

Reply:
aws-c-ops doesn't exist, and after thinking more on it, it's a necessary mechanism for any protocol development. Seems to make sense here to me. Http clients can use it, mqtt could technically etc....

Reply:
This just seems like the first of N misc parts we'll need to build a full SDK on the CRT that aren't big enough (like aws-c-auth) to warrant their own package.

What other parts will we be building in the next few months that **should** go into aws-c-ops vs aws-c-io?

Reply:
I think the line would be things that need more info/aggregation info than lives in aws-c-http, aws-c-io etc... So things like throttling, max conns, monitoring, metrics, etc.... I don't think aws-c-ops should just be the grab bag of misc components, but stuff that properly doesn't fit into the dependency graph as is. In this case the strategy interface doesn't really need more info than aws-c-io already provides. And any protocol stack can consume it as is. Another example would be, should this error result in a connection closure or reuse? The would need metadata that neither aws-c-io or aws-c-http contain.

Reply:
In fact, I can see multiple implementations of these being in a different lib. The interface though, makes most sense to me in aws-c-io. Just my thoughts at the moment.